src/lib/FreeC/IR/Syntax/Expr.hs

```diff
@@ -250,7 +256,10 @@ prettyExprPred' n expr@(TypeAppExpr _ e t _)
   | n <= 1 = prettyExprPred 1 e <+> char '@' <> prettyTypePred 2 t
   | otherwise = parens (prettyExprPred' 1 expr)
 prettyExprPred' n expr@(ErrorExpr _ msg _)
-  | n <= 1 = prettyString "error" <+> prettyString (show msg)
+  | n <= 1 = prettyString "error" <+> prettyString msg
```

Comment:
`show` must be called on `msg`, otherwise the message string is missing `"` in the output (same with the `Trace` case).
The `Trace` is also missing a specific case for visible type application resulting in the wrong order of arguments, i.e. currently `trace "Hello" x @a` is generated and should be `trace @a "Hello" x`.